dojo/tools/wapiti/parser.py

```diff
@@ -17,4 +33,100 @@ def __init__(self, file, test):
+                title = result.find('nvt/name').text
+                cve = result.find('nvt/cve').text
+                num_severity = result.find('nvt/risk_factor').text
+                severityfilter = Severityfilter()
+                severityfilter.eval_column(num_severity)
+                severity = severityfilter.severity
+                reference = result.find('nvt/xref').text
+                description = (result.find('description').text).encode('utf-8')
+                mitigation = "N/A"
+                impact = "N/A"
+
+                dupe_key = hashlib.md5(description + title + severity).hexdigest()
+
+                if dupe_key in self.dupes:
+                    finding = self.dupes[dupe_key]
+                    if finding.description:
+                        finding.description = finding.description
+                    self.process_endpoints(finding, host)
+                    self.dupes[dupe_key] = finding
+                else:
+                    self.dupes[dupe_key] = True
+
+                    finding = Finding(title=title,
+                                    test=test,
+                                    active=False,
+                                    verified=False,
+                                    cve=cve,
+                                    description=description,
+                                    severity=severity,
+                                    numerical_severity=Finding.get_numerical_severity(
+                                        severity),
+                                    mitigation=mitigation,
+                                    impact=impact,
+                                    references=reference,
+                                    dynamic_finding=True)
+
+                    self.dupes[dupe_key] = finding
+                    self.process_endpoints(finding, host)
+
+            self.items = self.dupes.values()
+
+    def process_endpoints(self, finding, host):
+        protocol = "http"
+        query = ""
+        fragment = ""
+        path = ""
+        url = urlparse(host)
+
+        if url:
+            path = url.path
+            if path == host:
+                path = ""
+
+        rhost = re.search(
+            r"(http|https|ftp)\://([a-zA-Z0-9\.\-]+(\:[a-zA-Z0-9\.&amp;%\$\-]+)*@)*((25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9])\.(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9]|0)\.(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9]|0)\.(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[0-9])|localhost|([a-zA-Z0-9\-]+\.)*[a-zA-Z0-9\-]+\.(com|edu|gov|int|mil|net|org|biz|arpa|info|name|pro|aero|coop|museum|[a-zA-Z]{2}))[\:]*([0-9]+)*([/]*($|[a-zA-Z0-9\.\,\?\'\\\+&amp;%\$#\=~_\-]+)).*?$",
+            host)
+        try:
+            protocol = rhost.group(1)
+            host = rhost.group(4)
+        except:
+            pass
+        try:
+            dupe_endpoint = Endpoint.objects.get(protocol="protocol",
```

Comment:
Why is protocol in quotation marks.
Did you put that on purpose ?

Reply:
Ohh didn't see that. let me check. this process endpoint function separate protocol, query, fragment, and path from URL and checks if dupe endpoints present or not and save endpoints. This function was used in Nikto scan parser to extract these from URL. I updated this function according to what I need. I think this is a mistake.